ultet/src/main/scala/za/co/absa/ultet/Ultet.scala

```diff
@@ -17,12 +17,41 @@
 package za.co.absa.ultet
 
 import scopt.OParser
+import za.co.absa.ultet.model.SQLEntry
 import za.co.absa.ultet.util.{CliParser, Config, DBProperties}
 
 import java.nio.file._
 import scala.collection.JavaConverters._
+import java.sql.{Connection, DriverManager, ResultSet}
+import scala.util.{Failure, Success, Try}
 
 object Ultet {
+  def runTransaction(connection: Connection, entries: Seq[SQLEntry]): Seq[ResultSet] = {
+    val autoCommitOriginalStatus = connection.getAutoCommit
+    connection.setAutoCommit(false)
+
+    val resultSets = Try {
+      entries.foldLeft(List[ResultSet]()) { case (acc, entry) =>
+        val statement = connection.createStatement()
+        statement.execute(entry.sqlExpression)
+        val ret: List[ResultSet] = acc :+ statement.getResultSet
```

Comment:
very minor in our case, but `statement.getResultSet :: acc` and at the end reversing the list should yield better algo complexity (hopefully my brain isn't dead already today)